File: sdkwork-backend-sdk-java/src/main/java/com/sdkwork/backend/api/CardApi.java
package com.sdkwork.backend.api;

import com.sdkwork.backend.http.HttpClient;
import com.sdkwork.backend.model.*;
import java.util.List;
import java.util.Map;

public class CardApi {
    private final HttpClient client;
    
    public CardApi(HttpClient client) {
        this.client = client;
    }

    /** Update an existing membership card */
    public PlusApiResultPlusCardVO update(PlusCardForm body) throws Exception {
        return (PlusApiResultPlusCardVO) client.put(ApiPaths.backendPath("/card"), body);
    }

    /** Create a new membership card */
    public PlusApiResultPlusCardVO create(PlusCardForm body) throws Exception {
        return (PlusApiResultPlusCardVO) client.post(ApiPaths.backendPath("/card"), body);
    }

    /** Update an existing card template */
    public PlusApiResultPlusCardTemplateVO updateTemplate(PlusCardTemplateForm body) throws Exception {
        return (PlusApiResultPlusCardTemplateVO) client.put(ApiPaths.backendPath("/card/template"), body);
    }

    /** Create a new card template */
    public PlusApiResultPlusCardTemplateVO createTemplate(PlusCardTemplateForm body) throws Exception {
        return (PlusApiResultPlusCardTemplateVO) client.post(ApiPaths.backendPath("/card/template"), body);
    }

    /** Get card templates by page */
    public PlusApiResultPagePlusCardTemplateVO createListByPage(QueryListForm body, Map<String, Object> params) throws Exception {
        return (PlusApiResultPagePlusCardTemplateVO) client.post(ApiPaths.backendPath("/card/template/list"), body, params);
    }

    /** Get all card templates */
    public PlusApiResultListPlusCardTemplateVO createListAllEntities(QueryListForm body) throws Exception {
        return (PlusApiResultListPlusCardTemplateVO) client.post(ApiPaths.backendPath("/card/template/list/all"), body);
    }

    /** Get membership cards by page */
    public PlusApiResultPagePlusCardVO createListByPageCard(QueryListForm body, Map<String, Object> params) throws Exception {
        return (PlusApiResultPagePlusCardVO) client.post(ApiPaths.backendPath("/card/list"), body, params);
    }

    /** Get all membership cards */
    public PlusApiResultListPlusCardVO createListAllEntitiesCard(QueryListForm body) throws Exception {
        return (PlusApiResultListPlusCardVO) client.post(ApiPaths.backendPath("/card/list/all"), body);
    }

    /** Get a membership card by ID */
    public PlusApiResultPlusCardVO getById(String id) throws Exception {
        return (PlusApiResultPlusCardVO) client.get(ApiPaths.backendPath("/card/" + id + ""));
    }

    /** Delete a membership card */
    public PlusApiResultBoolean delete(String id) throws Exception {
        return (PlusApiResultBoolean) client.delete(ApiPaths.backendPath("/card/" + id + ""));
    }

    /** Get a card template by ID */
    public PlusApiResultPlusCardTemplateVO getByIdTemplate(String id) throws Exception {
        return (PlusApiResultPlusCardTemplateVO) client.get(ApiPaths.backendPath("/card/template/" + id + ""));
    }

    /** Delete a card template */
    public PlusApiResultBoolean deleteTemplate(String id) throws Exception {
        return (PlusApiResultBoolean) client.delete(ApiPaths.backendPath("/card/template/" + id + ""));
    }
}

File: sdkwork-backend-sdk-kotlin/src/main/kotlin/com/sdkwork/backend/api/CardApi.kt
package com.sdkwork.backend.api

import com.sdkwork.backend.*
import com.sdkwork.backend.http.HttpClient

class CardApi(private val client: HttpClient) {

    /** Update an existing membership card */
    suspend fun update(body: PlusCardForm): PlusApiResultPlusCardVO? {
        return client.put(ApiPaths.backendPath("/card"), body) as? PlusApiResultPlusCardVO
    }

    /** Create a new membership card */
    suspend fun create(body: PlusCardForm): PlusApiResultPlusCardVO? {
        return client.post(ApiPaths.backendPath("/card"), body) as? PlusApiResultPlusCardVO
    }

    /** Update an existing card template */
    suspend fun updateTemplate(body: PlusCardTemplateForm): PlusApiResultPlusCardTemplateVO? {
        return client.put(ApiPaths.backendPath("/card/template"), body) as? PlusApiResultPlusCardTemplateVO
    }

    /** Create a new card template */
    suspend fun createTemplate(body: PlusCardTemplateForm): PlusApiResultPlusCardTemplateVO? {
        return client.post(ApiPaths.backendPath("/card/template"), body) as? PlusApiResultPlusCardTemplateVO
    }

    /** Get card templates by page */
    suspend fun createListByPage(body: QueryListForm? = null, params: Map<String, Any>? = null): PlusApiResultPagePlusCardTemplateVO? {
        return client.post(ApiPaths.backendPath("/card/template/list"), body, params) as? PlusApiResultPagePlusCardTemplateVO
    }

    /** Get all card templates */
    suspend fun createListAllEntities(body: QueryListForm? = null): PlusApiResultListPlusCardTemplateVO? {
        return client.post(ApiPaths.backendPath("/card/template/list/all"), body) as? PlusApiResultListPlusCardTemplateVO
    }

    /** Get membership cards by page */
    suspend fun createListByPageCard(body: QueryListForm? = null, params: Map<String, Any>? = null): PlusApiResultPagePlusCardVO? {
        return client.post(ApiPaths.backendPath("/card/list"), body, params) as? PlusApiResultPagePlusCardVO
    }

    /** Get all membership cards */
    suspend fun createListAllEntitiesCard(body: QueryListForm? = null): PlusApiResultListPlusCardVO? {
        return client.post(ApiPaths.backendPath("/card/list/all"), body) as? PlusApiResultListPlusCardVO
    }

    /** Get a membership card by ID */
    suspend fun getById(id: String): PlusApiResultPlusCardVO? {
        return client.get(ApiPaths.backendPath("/card/$id")) as? PlusApiResultPlusCardVO
    }

    /** Delete a membership card */
    suspend fun delete(id: String): PlusApiResultBoolean? {
        return client.delete(ApiPaths.backendPath("/card/$id")) as? PlusApiResultBoolean
    }

    /** Get a card template by ID */
    suspend fun getByIdTemplate(id: String): PlusApiResultPlusCardTemplateVO? {
        return client.get(ApiPaths.backendPath("/card/template/$id")) as? PlusApiResultPlusCardTemplateVO
    }

    /** Delete a card template */
    suspend fun deleteTemplate(id: String): PlusApiResultBoolean? {
        return client.delete(ApiPaths.backendPath("/card/template/$id")) as? PlusApiResultBoolean
    }
}
